### DISCUSSION TITLE
Could there be a `graphql-java-bom`?

### DISCUSSION THREAD

**Discussion by nmck257** [graphql-java/graphql-java#3677] [Ideas]
The core `graphql-java` library has a few companion libraries with aligned-but-not-matching version numbers.
It would be convenient and reduce risk of error if developers could import a BOM to define versions for all relevant dependencies.

See this discussion from the spring team, which mentions some of the benefits: https://github.com/spring-projects/spring-graphql/issues/242

**Comment by dondonz** [comment]
Hi thanks for the suggestion, let me discuss it with the team.

Yes I agree that the GraphQL Java version number isn't always matching for libraries such as extended scalars and validation, and Java Dataloader has another version numbering system altogether. I can see how that's confusing and inconvenient!

For others reading this thread, please do add your comments or add an upvote and that helps me work out what to prioritise.

**Reply by dondonz** [reply]
A colleague at work also asked about a BOM so that's 1 more upvote!
